src/app/services/auth/auth.service.ts

```diff
@@ -65,6 +68,16 @@ export class AuthService {
     }));
   }
 
+  setupPeriodicTokenGeneration(): void {
+    if (!this.generateTokenSubscription || this.generateTokenSubscription.closed) {
+      this.generateTokenSubscription = timer(0, 290 * 1000).pipe(
+        switchMap(() => this.generateToken()),
+        tap((token) => this.token2 = token),
+        untilDestroyed(this),
+      ).subscribe();
+    }
+  }
```

Comment:
I don't see any code that would clear this subscription. Wouldn't logging out and logging-in again result in two timers?